src/ds/d400/d400-options.h

```diff
@@ -188,4 +188,30 @@ namespace librealsense
 
         std::function<void(const option&)> _recording_function = [](const option&) {};
     };
+
+
+    class hid_sensor;
+    class gyro_sensitivity_option: public option_base
+    {
+    public:
+        gyro_sensitivity_option( const std::weak_ptr< hid_sensor > & sensor, const option_range & opt_range )
```

Comment:
Better to receive a shared_ptr (not reference) and inside set to a weak_ptr (non reference)

Reply:
Why? I don't agree.
Passing a shared_ptr by value will increase/decrease the reference counter unnecessarily.

Reply:
I think that the whole concept of smart pointers exist that users should not worry about the object lifetime.
The overhead should be small.
But we pay with performance yes.
I think it's a matter of style, so if you are sure that this function is called at the same context that you are OK, but if tomorrow someone will save this reference and use it in a different context, bad stuff can happen..